frontend/src/index.tsx

```diff
@@ -1,253 +1,66 @@
-        <img src={`data:image/png;base64,${item.image!}`} alt={item.name} class="h-20 w-20 rounded-xl object-cover" />
-        <div class="flex-grow-1 overflow-hidden">
-          <h2 class="text-xl font-semibold">{item.name}</h2>
-          <p class="max-width-text truncate text-gray-600">{item.description}</p>
-        </div>
-        <Switch>
-          <Match when={item.state === AppState.Initial}>
-            <button class="send-button bg-blue-500" onClick={onDownload}>
-              <div class="i-material-symbols:download text-white"></div>
-            </button>
-          </Match>
-          <Match when={item.state === AppState.Downloading}>
-            <div class="send-button bg-gray-500">
-              <div class="i-eos-icons:bubble-loading text-white"></div>
-            </div>
-          </Match>
-          <Match when={item.state === AppState.DownloadCancelled}>
-            <div class="send-button bg-red-500" >
-              <div class="i-material-symbols:error text-white"></div>
-            </div>
-          </Match>
-          <Match when={item.state === AppState.Received}>
-            <button class="send-button bg-green-500" onClick={onForward}>
-              <div class="i-material-symbols:forward text-white"></div>
-            </button>
-          </Match>
-          <Match when={item.state === AppState.Updating}>
-            <div class="i-eos-icons:bubble-loading text-black"></div>
-            <button class="send-button bg-green-500" onClick={onForward}>
-              <div class="i-material-symbols:forward text-white"></div>
-            </button>
-          </Match>
-        </Switch>
-      </div >
-      {
-        isExpanded() && (
-          <div class="flex flex-col">
-            <p class="my-4 text-gray-600">{item.description}</p>
-            <div class="my-2">
-              <p class="text-sm text-gray-600"><span class="font-bold"> Submitter: </span>{item.submitter_uri}</p>
-              <p class="break-all text-sm text-gray-600"><span class="font-bold"> Source code: </span>{item.source_code_url}</p>
-              <p class="text-sm text-gray-600"><span class="font-bold"> Version: </span>{item.version}</p>
-            </div>
-            {(item.state === AppState.Received || item.state === AppState.Updating) && <button class="self-center btn-gray" onClick={onRemove}>Remove from cache</button>}
-          </div>
-        )
-      }
-      <div class="mt-1 flex justify-center" onClick={() => setIsExpanded(!isExpanded())}>
-        <button class={`text-blue-800 ${isExpanded() ? 'i-carbon-up-to-top' : 'i-carbon-down-to-bottom'}`}>
-        </button>
-      </div>
-    </li >
-  )
-}
-
-interface AppListProps {
-  items: AppInfoWithState[]
-  search_query: string
-  onDownload: (id: string) => void
-  onForward: (id: string) => void
-  onRemove: (id: string) => void
+import type { StoreRequest } from '~/bindings/StoreRequest'
+
+const metaData = {
```

Comment:
naming this `metaData` feels kind of awkward, maybe @hpk42 has a better idea?

Reply:
it's fine for now i think.